extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/VertxHttpHotReplacementSetup.java

```diff
@@ -39,7 +40,8 @@ void handleHotReplacementRequest(RoutingContext routingContext) {
             routingContext.next();
             return;
         }
-        routingContext.vertx().executeBlocking(new Handler<Promise<Boolean>>() {
+        ConnectionBase connectionBase = (ConnectionBase) routingContext.request().connection();
```

Comment:
This change shouldn't be needed anymore.

Reply:
It is, the old code could resume on a different IO thread in the same vert.x instance, this makes sure it is the same thread that will be delivering the IO notifications.

Reply:
Sorry I can't see how, do you have a simple example in mind?

Reply:
OK